crates/turborepo-lib/src/package_graph/mod.rs

```diff
@@ -63,8 +64,44 @@ impl PackageGraph {
         PackageGraphBuilder::new(repo_root, root_package_json)
     }
 
-    pub fn validate(&self) -> Result<()> {
-        // TODO
+    pub fn validate(&self) -> Result<(), builder::Error> {
+        // This is equivalent to AcyclicGraph.Cycles from Go's dag library
+        let cycles = petgraph::algo::tarjan_scc(&self.workspace_graph)
```

Comment:
Aren't strongly connected components a subset of cycles?

Reply:
["A directed graph is acyclic if and only if it has no strongly connected subgraphs with more than one vertex"](https://en.wikipedia.org/wiki/Strongly_connected_component#Definitions). This is also exactly what we do in Go. `AcyclicGraph.Cycles` uses a Go implementation of Tarjan's SCC algorithm [source](https://github.com/pyr-sh/dag/blob/master/tarjan.go#L7)